### PR TITLE
Add feature to create and use parsed SQL queries cache files.

### PR DESCRIPTION
**Pull request by ienkovich**
The patch was done on top of the Config branch. New feature added only for ArrowBasedExecuteTest.

### CODE REVIEW

File: Tests/ArrowSQLRunner/RelAlgCache.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  void load();
+  void store() const;
+
+  struct CacheKey {
+    std::string sql;
+    int schema_id;
+    std::string user;
+    std::string db_name;
+    bool legacy_syntax;
+    bool is_explain;
```
Comment (alexbaden) [inline Tests/ArrowSQLRunner/RelAlgCache.h:60]: Ah, this is calcite explain -- so the payload is the rel alg tree string and not the serialized json rep of the rel alg?

Comment (ienkovich) [reply Tests/ArrowSQLRunner/RelAlgCache.h:60]: I simply use params to CalciteJNI::process call as a key and the resulting string as a value. This looks like the simplest way to make a cache and it should cover all kinds of queries, including explain queries and negative tests.

File: Tests/ArrowSQLRunner/RelAlgCache.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+           const std::string& schema_json,
+           const std::string& ra);
+
+  void load();
+  void store() const;
+
+  struct CacheKey {
+    std::string sql;
+    int schema_id;
+    std::string user;
```
[the remark this answers is not in the document]
Comment (alexbaden) [reply Tests/ArrowSQLRunner/RelAlgCache.h:57]: With JNI we can probably remove the user parameter. Previously a special elevated and protected session was created for calcite to read schemas from the database.

[the remark this answers is not in the document]
Comment (ienkovich) [reply Tests/ArrowSQLRunner/RelAlgCache.h]: Removed this arg.

**Review by alexbaden — APPROVED** [review]

**Pull request merged**